lib/config/rules/style.js

```diff
@@ -52,7 +52,7 @@ module.exports = {
   // Require or disallow a space before function parenthesis
   'space-before-function-paren': require('./rule-helpers/space-before-function-paren-rule'),
   // Checks variable names for various errors.
-  'variable-name': [true, 'ban-keywords', 'check-format', 'allow-leading-underscore'],
+  'variable-name': [true, 'ban-keywords', 'check-format', 'allow-pascal-case'],
```

Comment:
This `variable-rule` has limitations to it's flexibility. 
For the react config we would just want `PascalCase`. However with this rule, `"allow-pascal-case" allows PascalCase in addtion to camelCase.`.
Meaning if someone were to write their component in camel case, it would raise an error. 
I've created an issue to address the separation of these concerns.